Use move semantics for ShareableResource::map

donny-dont opened the pull request:
#### c62fe5aabb3816c7590b09a775b398269c27a5e3
<pre>
Use move semantics for ShareableResource::map
<a href="https://bugs.webkit.org/show_bug.cgi?id=256062">https://bugs.webkit.org/show_bug.cgi?id=256062</a>

Reviewed by Kimmo Kinnunen.

Continuation of 263362@main which used move semantics for
`ShareableResource::Handle` to signify transferring ownership.

* Source/WebKit/NetworkProcess/cache/NetworkCacheEntry.cpp:
* Source/WebKit/Shared/ShareableResource.cpp:
* Source/WebKit/Shared/ShareableResource.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
* Source/WebKit/WebProcess/Network/NetworkProcessConnection.cpp:
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:

Canonical link: <a href="https://commits.webkit.org/263683@main">https://commits.webkit.org/263683@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b54d2f50797aaac2800257731b5e017f84286d47

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5341 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5478 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5666 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6878 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5382 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5705 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5459 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/6061 "2 flakes 106 failures") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5440 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5519 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4787 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6916 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2998 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/11957 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4864 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4871 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6509 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5296 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4353 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4761 "Built successfully") | | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1301 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8858 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebKit/Shared/ShareableResource.h:
```diff
@@ -50,7 +50,7 @@ class ShareableResourceHandle {
     static WARN_UNUSED_RETURN bool decode(IPC::Decoder&, ShareableResourceHandle&);
 
-    RefPtr<WebCore::SharedBuffer> tryWrapInSharedBuffer() const;
+    RefPtr<WebCore::SharedBuffer> tryWrapInSharedBuffer();
```
So also this could be marked as rvalue reference, so that the caller doesn't end up with the contents of `*this` mutated behind the scenes.
So you'd mark this up as being rvalue qualified function:
```
    RefPtr<WebCore::SharedBuffer> tryWrapInSharedBuffer() &&;
```

Then caller would call it like:
```
        m_buffer = WTFMove(m_shareableResourceHandle).tryWrapInSharedBuffer();
``

donny-dont committed:
Use move semantics for ShareableResource::map
https://bugs.webkit.org/show_bug.cgi?id=256062

Reviewed by Kimmo Kinnunen.

Continuation of 263362@main which used move semantics for
`ShareableResource::Handle` to signify transferring ownership.

* Source/WebKit/NetworkProcess/cache/NetworkCacheEntry.cpp:
* Source/WebKit/Shared/ShareableResource.cpp:
* Source/WebKit/Shared/ShareableResource.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
* Source/WebKit/WebProcess/Network/NetworkProcessConnection.cpp:
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:

Canonical link: https://commits.webkit.org/263683@main
